Kernel/Devices/PCISerialDevice.cpp

```diff
@@ -21,12 +24,21 @@ void PCISerialDevice::detect()
             if (board_definition.device_id != id)
                 continue;
 
+            auto maybe_new_baud = SerialDevice::termios_baud_from_serial(board_definition.baud_rate);
+            if (!maybe_new_baud.has_value()) {
+                dbgln("FIXME: PCISerialDevice's speed is missing from termios");
+                continue;
```

Comment:
this should be a return, not a continue, we already know this is the matching board definition, so theres no point to iterating over the others